sqs/sqs.go

```
@@ -3,80 +3,96 @@ package sqs
+	MaximumMessageSize            string                    `json:"MaximumMessageSize,omitempty"`
+	MessageRetentionPeriod        string                    `json:"MessageRetentionPeriod,omitempty"`
+	Policy                        string                    `json:"Policy,omitempty"`
+	ReceiveMessageWaitTimeSeconds string                    `json:"ReceiveMessageWaitTimeSeconds,omitempty"`
+	RedrivePolicy                 string                    `json:"RedrivePolicy,omitempty"`
+	DeadLetterTargetArn           string                    `json:"deadLetterTargetArn,omitempty"`
+	MaxReceiveCount               string                    `json:"maxReceiveCount,omitempty"`
+	VisibilityTimeout             string                    `json:"VisibilityTimeout,omitempty"`
+	KmsMasterKeyId                string                    `json:"KmsMasterKeyId,omitempty"`
+	KmsDataKeyReusePeriodSeconds  string                    `json:"KmsDataKeyReusePeriodSeconds,omitempty"`
+	SqsManagedSseEnabled          string                    `json:"SqsManagedSseEnabled,omitempty"`
+	FifoQueue                     QueueConfigAttributesBool `json:"FifoQueue,omitempty"`
+	ContentBasedDeduplication     QueueConfigAttributesBool `json:"ContentBasedDeduplication,omitempty"`
+	DeduplicationScope            string                    `json:"DeduplicationScope,omitempty"`
+	FifoThroughputLimit           string                    `json:"FifoThroughputLimit,omitempty"`
 }
 
-func (q QueueConfigAtrributes) Attributes() map[string]string {
-	b, _ := json.Marshal(q)
+// QueueConfigAttributesBool is a custom type for bool values in QueueConfigAttributes
+// that supports marshaling to string.
+type QueueConfigAttributesBool bool
+
+func (q QueueConfigAttributesBool) MarshalText() ([]byte, error) {
+	if q {
+		return []byte("true"), nil
+	}
+	return []byte("false"), nil
+}
+
+func (q QueueConfigAttributes) Attributes() (map[string]string, error) {
```

i started to return errors from here as it was silently failing earlier 🙈